### DISCUSSION TITLE
convert video file into contact-sheet

### DISCUSSION THREAD

**Discussion by mhdeditor** [ImageMagick/ImageMagick#6853] [Help]
### ImageMagick version

 7.1.1-21

### Operating system, version and so on

macOS Ventura 13.5.2 (22G91)

### Description

Hi,

I’m looking a way to :

From a .mov or .mp4 file : extraction of 1 image per 2 secondes (could be an ajustable amount of secondes ?)
Scale images to :  width = 200 px (could be scalable ?) h= respect aspect ratio 
Edit images in time order on one file (could be scalable ?) contact-sheet (.jpeg or pdf).  No border between images.

This is in order to show students an overview of the edit of movie ! It could be very helpful !

I have installed  ffmpeg (version  version 6.0) and imagemagick (version  7.1.1-21), but coding such a script is far away my capacities !

Many thanks
Kind regards

**Comment by snibgo** [comment]
Perhaps all the work can be done with ffpeg, but I'm not an expert at that.

One method is:
```
ffmpeg -i in.mp4 -s 200x200 -r 2 x%06d.png
```
This sets the output frame rate to 2 fps (frames per second). If the input frame rate is 25 fps, the output will have 1/12.5 of the input frames. Adjust to whatever you want.

The output size of each frame will be 200x200 pixels. Again, adjust to whatever you want.

For more details about ffmpeg, consult the ffmpeg documentation.

Then:
```
magick montage x*.png -geometry +0+0 -tile 5x out.png
```
This makes a contact sheet 5 frames wide.

You could use jpg instead of png for smaller filesizes.

**Comment by GeeMack** [comment]
Okay, this is an ImageMagick discussion forum, but I can get you started with a "ffmpeg" command to extract all the images from a video at 30 second intervals, resize them to 200 pixels wide, and keep the aspect ratio...

    ffmpeg -i input.mp4 -filter_complex "fps=1/30,scale=200:-1" output_%04d.png

To select the interval in seconds, change the "30" in "fps=1/30" to the number of seconds you want between images.

The output images will be PNG format, and named "output_NNNN.png", where NNNN is 4 digits with leading zeros. If you anticipate more than 9,999 images, change the "4" in "%04d" to the number of digits that you'll need in the filenames.

It should be a fairly simple matter to use ImageMagick's "montage" tool to assemble the results into your contact sheets, but it would be more helpful to know what sort of layout of rows, columns, number of pages, etc. you have in mind.  See **snibgo**'s reply for some direction with that.

**Reply by mhdeditor** [reply]
Hi Snibgo, hi Geemack,

Thanks a lot for your replies.

Ffmpeg commande works great ! I try with a short excerpt and extract 30 images. 

For the second step, the montage, I can be more precise. 
The idea could be : 
Choose a number of rows and a number of columns. Then imagemagick generate a .jpg file. The dimension of this file is based on the number of row and column, the size of thumbnails and the number of thumbnails. 4 parameters for an adaptable file dimension.

Thanks !

**Comment by snibgo** [comment]
> Choose a number of rows and a number of columns.

So that would be "`-tile 5x6`" if you want up to 5 columns and 6 rows in each output image.

> Then imagemagick generate a .jpg file.

If there are more than 30 frames (5 x 6 = 30) then IM will create multiple output images. So you would get multiple jpg files, or a single pdf with multiple pages.

**Comment by mhdeditor** [comment]
I'm sorry, in
magick montage x*.png -geometry +0+0 -tile 5x6 out.png

I did not attempt to introduce my array of 30 images in place of x*png
I did different try, i guess it's basic, but...

thx

**Comment by snibgo** [comment]
I don't understand your question.

What ffmpeg command did you use? Did it make PNG files? What are they named?

**Comment by mhdeditor** [comment]
ffmpeg -i /Users/MxWORK/Desktop/poster/FAR.mov -filter_complex "fps=1/10,scale=200:-1" /Users/MxWORK/Desktop/poster/exportffmpeg-output_%04d.jpg

and I get 30 images named "exportffmpeg-output_0001.jpg" ....  "exportffmpeg-output_0030.jpg"

I assume I use .jpg and will take in account in the IM command

**Comment by snibgo** [comment]
Your `magick` command could be:
```
magick montage /Users/MxWORK/Desktop/poster/exportffmpeg-output_*.jpg -geometry +0+0 -tile 5x6 out.png
```
Instead of `out.png`, use whatever filename (including extension) you want.

**Reply by mhdeditor** [reply]
no output file generates...

**Comment by mhdeditor** [comment]
my mistake it's ok !

**Comment by mhdeditor** [comment]
Thanks a lot to you sigbo and GeeMack for guiding me in that universe I'm discovering !

As a summ up of the travel, see below the commands that work for my purpose. 
Juste in case some people want to achieve the same trick.
If the scripting is not academic you will correct it please.

Thx again ! 
best wishes

This exemple assume the extract of 1 image per 5 seconde, 
the film based on is 3 hours lenght and generate 2168 images (!)
using -tile 40x120, IM generates a biii!!!g dimension file (wich is what I wanted)

Here is the code 

ffmpeg -i /Users/YOUR PATH/filename.mp4 -filter_complex "fps=1/5,scale=200:-1" /Users/YOUR PATH/output_%04d.jpg

magick montage /Users/YOUR PATH/output_*.jpg -geometry +0+0 -tile 40x120  /Users/YOUR PATH/out.jpeg
